.github/workflows/e2e.yml

```diff
@@ -61,6 +62,13 @@ jobs:
           test-cloud-shell:$GITHUB_SHA
       env:
         PROJECT_ID: ${{ secrets.E2E_PROJECT_ID  }}
+    - name: Run Monitoring Integration Tests
+      run: |
+        set -x
+        # install dependencies
+        python3 -m pip install -r tests/requirements.txt
+        # run tests
+        python3 tests/monitoring_integration_test.py ${{ secrets.E2E_PROJECT_ID }}
```

Comment:
You actually have to explicitly attach the secret to the step. See the step above yours:

```
      env:
        PROJECT_ID: ${{ secrets.E2E_PROJECT_ID  }}
```

You can then use PROJECT_ID as an environment variable within the step

Reply:
Also, you don't seem to need the id in your script, only the name. if you send it in as projects/$PROJECT_ID, you can avoid storing 2 variables

Reply:
Fixed! thanks for pointing this out :)